cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -240,22 +231,21 @@ def test_provision_approval(appliance, provider, vm_name, smtp_test, request,
                                                                  provider).cleanup_on_provider()
                      for name in vm_names]
         )
-    subject = VersionPicker({
-        LOWEST: "your virtual machine configuration was approved",
-        "5.10": "your {} request was approved".format(vm_type)
-    }).pick()
+
     try:
-        wait_for(
-            lambda:
-            len(filter(
-                lambda mail:
-                normalize_text(subject) in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == 1,
-            num_sec=120, delay=5)
+        wait_for(lambda: len(smtp_test.get_emails()) >= 2,
```

Comment:
Why does `smtp_test.get_emails()` need to be called twice? Is there a reason it isn't stored as a variable?

Reply:
We're calling it in the wait_for loop, but inside of the lambda, so wait_for isn't returning it when the wait is completed.  I have to call it again to actually get the contents of its return.